### PR TITLE
Feat/residents modules

### PR DESCRIPTION
**Pull request by vsantos1711**

### CODE REVIEW

File: back/src/application/residents/dto/create-resident.dto.ts
Comment (mathmartins2) [inline back/src/application/residents/dto/create-resident.dto.ts:1]: add min and max to name length
add string validation to condominiumId
add is optional to email
add min and max buildingApartment

File: back/src/application/residents/resident.service.ts
Comment (mathmartins2) [inline back/src/application/residents/resident.service.ts:1]: remove space from 12 line

**Review by mathmartins2 — CHANGES_REQUESTED** [review]

File: back/src/application/residents/resident.service.ts
Comment (vsantos1711) [reply back/src/application/residents/resident.service.ts:1]: Changes maked!

File: back/src/application/residents/dto/create-resident.dto.ts
Comment (vsantos1711) [reply back/src/application/residents/dto/create-resident.dto.ts:1]: Everything in order!

**Review by mathmartins2 — APPROVED** [review]

**Pull request merged**